include/cocaine/framework/worker/receiver.hpp

```diff
@@ -22,29 +22,36 @@
 #include <boost/optional/optional.hpp>
 
 #include <cocaine/forwards.hpp>
+// #include <cocaine/hpack/header.hpp>
 
 #include "cocaine/framework/forwards.hpp"
 
 namespace cocaine {
-
 namespace framework {
-
 namespace worker {
 
+struct frame_t {
+    std::string data;
+    // hpack::header_storage_t meta;
+};
+
 class receiver {
     std::shared_ptr<basic_receiver_t<worker_session_t>> session;
 
 public:
-    /*!
-     * \note this constructor is intentionally left implicit.
-     */
+    /// \note this constructor is intentionally left implicit.
     receiver(std::shared_ptr<basic_receiver_t<worker_session_t>> session);
 
-    auto recv() -> task<boost::optional<std::string>>::future_type;
+    template<typename R = std::string>
+    auto recv() -> future<boost::optional<R>>;
```

Comment:
Because I can. In normal languages it is called **type inference** depending on expected result type.